website/src/documentation/programming-guide.md

```diff
@@ -3121,3 +3121,515 @@ public class MyMetricsDoFn extends DoFn<Integer, Integer> {
   }
 }
 ```  
+
+## 10. State and Timers {#state-and-timers}
+Beam's windowing and triggering facilities provide a powerful abstraction for grouping and aggregating unbounded input
+data based on timestamps. However there are aggregation use cases for which Beam's windows and triggers are not the best
+fit. Beam provides an API for manually managing per-key state, allowing for finer-grained control over aggregations.
+
+Beam's state API models state per key. To use the state API, you start out with a keyed `PCollection`, which in Java
+is modeled as a `PCollection<KV<K, V>>`. A `ParDo` processing this `PCollection` can now declare state variables. Inside
+the `ParDo` these state variables can be used to write or update state for the current key or to read previous state
+written for that key. State is always fully scoped only to the current processing key.
+
+Windowing can still be used together with stateful processing. All state for a key is scoped to the current window. This
```

Comment:
Also important to point out that you can use different window strategies before the State call. Something which I have seen folks not realise until pointed out.
For example Window+Combiner as a reducer before data comes to the state DoFn

Reply:
done